src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+package mate.academy.dao.impl;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String FIND_ALL_QUERY = "SELECT * FROM book;";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM book WHERE id = ?;";
+    private static final String CREATE_QUERY = "INSERT INTO book (title, price) VALUES (?, ?);";
+    private static final String UPDATE_QUERY = "UPDATE book SET title = ?, price = ? WHERE id = ?;";
+    private static final String DELETE_QUERY = "DELETE FROM book WHERE id = ?;";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(CREATE_QUERY,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Tried to create a book but failed to do so");
+            }
+
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create a book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = new Book();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                book = mapResultSetToBook(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to find the book with an id of "
+                    + id, e);
+        }
+        return Optional.of(book);
```

Comment:
When `findById` does not find a book, it should not return an empty `Book` object inside an `Optional`. It should return `Optional.empty()` instead.